packages/expo-device/ios/UIDevice.swift

```diff
@@ -0,0 +1,261 @@
+import UIKit
+
+struct ExpoDeviceType {
+  let modelName: String
+  let deviceYearClass: Int?
+}
+
+public extension UIDevice {
+  // Credit: https://stackoverflow.com/a/26962452
+  static let modelIdentifier: String = {
+    var systemInfo = utsname()
+    uname(&systemInfo)
+    let machineMirror = Mirror(reflecting: systemInfo.machine)
+    let identifier = machineMirror.children.reduce("") { identifier, element in
+      guard let value = element.value as? Int8, value != 0 else {
+        return identifier
+      }
+      return identifier + String(UnicodeScalar(UInt8(value)))
+    }
+
+    return identifier
+  }()
+
+  // swiftlint:disable function_body_length
```

Comment:
Hmm, looks like it now complains about `closure_body_length` rule 😂 
```suggestion
  // swiftlint:disable function_body_length closure_body_length
```